src/main.rs

```diff
@@ -146,10 +149,26 @@ pub(crate) extern "C" fn main(hart_id: usize, device_tree_blob_addr: usize) -> !
 }
 
 fn main_loop(ctx: &mut VirtContext, mctx: &mut MiralisContext) -> ! {
+    #[cfg(any(feature = "benchmark-instr", feature = "benchmark-time"))]
+    let mut bctx = BenchmarkContext::new();
+
     loop {
         unsafe {
+            #[cfg(any(feature = "benchmark-instr", feature = "benchmark-time"))]
```

Comment:
Try to avoid using `cfg`s, they should be used as a last resort because they quickly creep-up and makes Rust tooling much less powerful. Typically the type-checker will not check statements under deactivated `cfg`s, so we could commit code that doesn't compile on the main branch for instance, which is pretty bad.

In terms of performance there is no difference with using an `if` statement with a constant conditions (as we could get using our existing configuration system): the compiler will optimize the dead code away.